aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -134,70 +201,110 @@ static void upsample_nearest2d_out_cuda_template(
   int input_height = input_.size(2);
   int input_width = input_.size(3);
 
-  Tensor input = input_.contiguous();
+  const float height_scale = compute_scales_value<float>(scales_h, input_height, output_height);
+  const float width_scale = compute_scales_value<float>(scales_w, input_width, output_width);
 
-  if (input.numel() == 0) {
-    return;
-  }
+  const auto memory_format = input_.suggest_memory_format();
+
+  if (memory_format == at::MemoryFormat::ChannelsLast) {
+    TORCH_INTERNAL_ASSERT(output.suggest_memory_format() == memory_format, "output should also be channels-last memory format");
```

Comment:
this should be `.contiguous(MemoryFormat::ChannelsLast)` rather than just `suggest_memory_format`